tokio/src/sync/notify.rs

```diff
@@ -397,6 +398,28 @@ pub struct Notified<'a> {
 unsafe impl<'a> Send for Notified<'a> {}
 unsafe impl<'a> Sync for Notified<'a> {}
 
+/// Future returned from [`Notify::notified_owned()`].
+///
+/// This future is fused, so once it has completed, any future calls to poll
+/// will immediately return `Poll::Ready`.
+#[derive(Debug)]
+#[must_use = "futures do nothing unless you `.await` or poll them"]
+pub struct NotifiedOwned {
```

Comment:
This is new type, we also need to add tests in `tokio/tests/async_send_sync.rs`.